tests/std/tests/Dev10_881629_vector_erase_return_value/test.cpp

```diff
@@ -31,11 +31,12 @@ void test_case_devcom_776568() {
     c.emplace_back(1);
     c.emplace_back(2);
     c.emplace_back(3);
-    const auto after = c.begin() + 1;
-    auto it          = c.erase(c.begin(), c.begin());
-    assert(after->i == 2); // asserts that the iterator was not invalidated by iterator debugging
-    assert(it == c.begin());
+    const auto after = c.begin() + 2;

```

Comment:
> The previous test case test deque::erase(begin, begin) did not test self-assignment. Because in that special case the issue mentioned in #1118 wasn't happening. I've changed it to erase(begin+1, begin+1) which now actually can test for avoiding self-assignment.

Aha, thanks for finding why the tests missed this!

> however, I wasn't sure that should I just modify the test case or add a new test block?

I think that it would be better to add a new block of tests, instead of changing the existing coverage, which was developed for a bug in `vector`. (The erase(begin, begin) scenario might have been important for `vector`, or it might be important in the future.) Two options:

1. Wrap the original contents of this function `test_case_devcom_776568()` in braces, then add a new set of braces with your erase(begin+1, begin+1) test coverage. I recommend adding a comment `// Also test GH-1118` in this case.

2. Add a new function `test_case_gh_1118()`, and then call it for `vector` and `deque` at the end of `main()`.

Either option is fine.

Reply:
Thank you @StephanTLavavej . I've preferred the second option. I've added a new function called `test_case_gh_1118()` for this case.